File: 4/4a.js
const fs = require('fs');

const input = fs.readFileSync('./4.txt', 'utf8');

const pairs = input.split(/\r?\n/);

let res = 0;

for (const pair of pairs) {
  const elves = pair
    .split(',')
    .map((elf) => elf.split('-').map((x) => parseInt(x, 10)));

  if (
    (elves[0][0] >= elves[1][0] && elves[0][1] <= elves[1][1]) ||
    (elves[0][0] <= elves[1][0] && elves[0][1] >= elves[1][1])
  ) {
    res += 1;
  }
}

console.log(res);

File: 4/4a.py
with open("4.txt") as file:
    pairs = file.read().split('\n')
    res = 0
    # for pair in pairs:
    #     elves = [int(x) for elf in pair.split(',') for x in elf.split('-')]
    for pair in pairs:
        elves = [[int(x) for x in elf.split('-')] for elf in pair.split(',')]

        if ((elves[0][0] >= elves[1][0] and elves[0][1] <= elves[1][1]) or
                (elves[0][0] <= elves[1][0] and elves[0][1] >= elves[1][1])):
            res += 1
    print(res)
